core/sds-aspect-model-document-generators/src/main/java/io/openmanufacturing/sds/aspectmodel/generator/AspectModelHelper.java

```diff
@@ -0,0 +1,180 @@
+        if (entities != null) {
+            entities.sort(Comparator.comparing(entity -> entity.getPreferredName(local)));
+        }
+        return entities;
+    }
+
+    public static List<Operation> sortOperationsByPreferredName(final List<Operation> operations, final Locale local) {
+        if (operations != null) {
+            operations.sort(Comparator.comparing(operation -> operation.getPreferredName(local)));
+        }
+        return operations;
+    }
+
+    public static List<Entity> getEntities(final Aspect aspectModel) {
+        final List<Property> properties = getProperties(aspectModel);
+        return properties.stream()
+                .filter(property -> property.getDataType().isPresent() && property.getDataType().get() instanceof Entity
+                        || property.getCharacteristic() instanceof Either)
+                .map(property -> {
+                    if (property.getCharacteristic() instanceof Either) {
+                        Either either = (Either) property.getCharacteristic();
+                        List<Entity> eitherEntities = new ArrayList<>();
+                        if (either.getLeft().getDataType().isPresent()) {
+                            eitherEntities.add((Entity) either.getLeft().getDataType().get());
+                        }
+                        if (either.getRight().getDataType().isPresent()) {
+                            eitherEntities.add((Entity) either.getRight().getDataType().get());
+                        }
+                        return eitherEntities;
+                    }
+                    return Arrays.asList((Entity) property.getDataType().get());
+                })
+                .flatMap(List::stream)
+                .distinct()
+                .collect(Collectors.toList());
+    }
+
+    public static List<Property> getProperties(final Aspect aspectModel) {
+        final List<Property> properties = new ArrayList<>(aspectModel.getProperties());
+        aspectModel.getOperations().forEach(operation -> properties.addAll(getProperties(operation)));
+        final Set<Property> characteristicProperties = new HashSet<>();
+        properties
+                .forEach(property -> characteristicProperties.addAll(getProperties(property.getCharacteristic())));
+        properties.addAll(characteristicProperties);
+        return properties;
+    }
+
+    public static Set<Property> getProperties(final Operation operation) {
+        return Stream.concat(operation.getInput().stream(), operation.getOutput().stream())
+                .collect(Collectors.toSet());
+    }
+
+    public static Set<Constraint> getConstraints(final Property property) {
+        final Set<Constraint> constraints = new HashSet<>();
+        if (property.getCharacteristic() instanceof Trait) {
+            constraints.addAll(((Trait) property.getCharacteristic()).getConstraints());
+        }
+        return constraints;
+    }
+
+    @SuppressWarnings("squid:S2250")
```

Comment:
Please add a comment to the annotation that explains what is suppressed and why

Reply:
done